batavia/core/exceptions.js

```diff
@@ -22,16 +22,64 @@ BaseException.prototype.toString = function() {
 }
 
 BaseException.prototype.__str__ = function() {
-    if (this.msg) {
-        return this.msg
+    if (this.args.length) {
+        if (this.args.length === 1) {
+            return this.args[0].toString()
+        }
+        // Multiple args. Format like a tuple.
+        let result = '(' + this.args[0].toString()
+        for (let arg in this.args.slice(1)) {
+            result += ', ' + arg.toString()
```

Comment:
That expected failure on the new test was niggling at me so I took at look. It's the very unfortunate JS `for...in` vs `for...of` that is primary cause, but additionally, it looks like BaseException's `__str__` calls `__repr__` on the args when there are multiple instead of `__str__`. You could probably move the parse function you defined in `__repr__` and reuse that.

```
>>> class Blah:
...   def __str__(self):
...     return "StrBlah"
...   def __repr__(self):
...     return "ReprBlah"
...
>>> str(ValueError(Blah(), Blah()))
'(ReprBlah, ReprBlah)'
>>> str(ValueError(Blah()))
'StrBlah'
```

Reply:
Very nice clean up for this, using the tuple code.